cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/elements/hvdc/Adjacency.java

```diff
@@ -36,23 +37,25 @@ enum AdjacentType {
 
     private final Map<String, List<Adjacent>> adjacency;
 
-    Adjacency(CgmesModel cgmesModel, AcDcConverterNodes acDcConverterNodes) {
+    Adjacency(CgmesModel cgmesModel, Context context, AcDcConverterNodes acDcConverterNodes) {
         adjacency = new HashMap<>();
         cgmesModel.dcLineSegments().forEach(dcls -> computeDcLineSegmentAdjacency(cgmesModel, dcls));
 
         acDcConverterNodes.getConverterNodes()
             .forEach((key, value) -> computeAcDcConverterAdjacency(value.acNode,
                 value.dcNode));
 
-        cgmesModel.groupedTransformerEnds().forEach((t, ends) -> {
-            if (ends.size() == 2) {
-                computeTwoWindingsTransformerAdjacency(cgmesModel, ends);
-            } else if (ends.size() == 3) {
-                computeThreeWindingsTransformerAdjacency(cgmesModel, ends);
-            } else {
-                throw new PowsyblException(String.format("Unexpected TransformerEnds: ends %d", ends.size()));
-            }
-        });
+        cgmesModel.transformers().stream()
+                .map(t -> context.transformerEnds(t.getId("PowerTransformer")))
```

Comment:
Could you use `CgmesNames.POWER_TRANSFORMER`? It could prevent typos.

Reply:
Sorry, I was not enough precise.
`"PowerTransformer"` is also used in the following classes:
- `Conversion`
- `NodeContainerMapping`
- `PhaseAngleClock`
- `NodeEquipment`

Could you replace it in these too?

Reply:
Seen with @rcourtier: the `"PowerTransformer"` string here is the name of a variable in the property bag, but it is not required that it matches the CGMES class name. These are different concepts and using the same constant may be confusing.
The previous state could be restored (something better could be done, but it requires a global refactoring that could be done later).